apps/camera/js/lib/camera/focus.js

```diff
@@ -81,6 +81,7 @@ Focus.prototype.configureFocusModes = function(mode) {
   this.faceDetection =
     faceDetectionUserEnabled && faceDetectionSupported && mode === 'picture';
   this.mozCamera.addEventListener('focus', this.onAutoFocusStateChange);
+  this.focusState = undefined;
```

Comment:
Again, I don't see how this fits in with updateFocusState. The only way I see a problem is when you reconfigure the focus (i.e. due to a camera reconfigure / switch), the cached focus state is now wrong. We should be clearing it. But that should happen in pause() rather than configureFocusModes().

Also a unit test case that fails now and passes with your change (add to test/unit/lib/camera/focus_test.js) would make this more clear.